bull-transformer-generator/transformer-bytecode-adapter/src/test/java/com/hotels/beans/generator/bytecode/TransformerBytecodeAdapterTest.java

```diff
@@ -0,0 +1,184 @@
+                .spec(spec)
+                .build();
+    }
+
+    @Test
+    public void shouldCreateANewTransformerWithGivenSpec() {
+        // WHEN
+        var actual = underTest.newTransformer(Source.class, Destination.class);
+
+        // THEN
+        assertNotNull(actual, "a new Transformer instance is never null");
+        then(spec).should().build(Source.class, Destination.class);
+    }
+
+    @Test
+    public void shouldCreateANewTransformerWithDefaultPackage() {
+        // WHEN
+        var actual = underTest.newTransformer(Source.class, Destination.class);
+
+        // THEN
+        assertThat(actual.getClass().getName(), startsWith(DEFAULT_PACKAGE));
+    }
+
+    @Test
+    public void shouldCreateANewTransformerWithGivenPackage() {
+        // GIVEN
+        var packageName = "foo.bar";
+        underTest = TransformerBytecodeAdapter.builder()
+                .packageName(packageName)
+                .spec(spec)
+                .build();
+
+        // WHEN
+        var actual = underTest.newTransformer(Source.class, Destination.class);
+
+        // THEN
+        assertThat(actual.getClass().getName(), startsWith(packageName));
+    }
+
+    @Test(expectedExceptions = RuntimeException.class)
+    public void shouldThrowRuntimeExceptionIfCompilerFails() throws Exception {
+        // GIVEN
+        CachedCompiler compiler = given(mock(CachedCompiler.class).loadFromJava(anyString(), anyString()))
+                .willThrow(ClassNotFoundException.class)
+                .getMock();
+        underTest = TransformerBytecodeAdapter.builder()
+                .compiler(compiler)
+                .spec(spec)
+                .build();
+
+        // WHEN
+        underTest.newTransformer(Source.class, Destination.class);
+    }
+
+    @Test(dataProvider = "nonCompliantTransformers", expectedExceptions = RuntimeException.class)
+    public void shouldThrowRuntimeExceptionIfTransformerIsNonCompliant(final Class<? extends Transformer> trClass)
+            throws Exception {
+        // GIVEN
+        underTest = TransformerBytecodeAdapter.builder()
+                .compiler(mockCompilerWith(trClass))
+                .spec(spec)
+                .build();
+
+        // WHEN
+        underTest.newTransformer(Source.class, Destination.class);
+    }
+
+    @DataProvider
+    public static Object[][] nonCompliantTransformers() {
+        return new Object[][]{
+                {TransformerPrivateCtor.class},
+                {TransformerCtorWithArgs.class},
+                {TransformerCtorThrows.class}
+        };
+    }
+
+    private static CachedCompiler mockCompilerWith(final Class<? extends Transformer> trClass)
+            throws ClassNotFoundException {
+        return given(mock(CachedCompiler.class).loadFromJava(anyString(), anyString()))
+                .willReturn(trClass)
+                .getMock();
+    }
+
+    //********************************************************
```

Comment:
please move those classes in separated files

Reply:
done

Reply:
Oh, I thought it would be nicer to keep them together with the test since they're not going to be used elsewhere and to avoid increasing the files count. But if there's a policy of "one class per file", it's fine.